From 6c8afab683e24722d48105c5ee18c7b3cacd338f Mon Sep 17 00:00:00 2001
From: "Joel Fernandes (Google)" <joel@joelfernandes.org>
Date: Sun, 16 Oct 2022 16:22:59 +0000
Subject: [PATCH 07/20] FROMGIT: rcu/sync: Use call_rcu_hurry() instead of
 call_rcu

call_rcu() changes to save power will slow down rcu sync. Use the
call_rcu_hurry() API instead which reverts to the old behavior.

(cherry picked from commit 438855f961dd8f7b1350a5d533ff7dc7adc79fb5
 https://git.kernel.org/pub/scm/linux/kernel/git/paulmck/linux-rcu.git rcu/next)

Bug: 258241771
Bug: 222463781
Test: powerIdle lab tests.
Signed-off-by: Joel Fernandes (Google) <joel@joelfernandes.org>
Change-Id: Ibecdb2c54b2978d856425a912b5107d4fd144512
Signed-off-by: Joel Fernandes <joelaf@google.com>
Reviewed-on: https://chromium-review.googlesource.com/c/chromiumos/third_party/kernel/+/3965075
Reviewed-by: Ross Zwisler <zwisler@google.com>
(cherry picked from commit c2c336c9d59fa3e62ee64270ddc27b1e86fada0b)
[Cherry picked from chromeos-5.10 tree. Minor tweaks to commit message
to match Android style]
Signed-off-by: Qais Yousef <qyousef@google.com>

Upstream: 7651d6b25086656eacfdd8356bfe3a21c0c2d79d ("rcu/sync: Use call_rcu_hurry() instead of call_rcu")
ACK android13-5.10: d9c3b8f4f822025b4c69b17d4800a7349ef193b5, taken from arter97/master
Backport notes: call_rcu_flush() renamed to call_rcu_hurry().
---
 kernel/rcu/sync.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/rcu/sync.c b/kernel/rcu/sync.c
index d4558ab7a07d..bbb384feb85b 100644
--- a/kernel/rcu/sync.c
+++ b/kernel/rcu/sync.c
@@ -44,7 +44,7 @@ static void rcu_sync_func(struct rcu_head *rhp);
 
 static void rcu_sync_call(struct rcu_sync *rsp)
 {
-	call_rcu(&rsp->cb_head, rcu_sync_func);
+	call_rcu_hurry(&rsp->cb_head, rcu_sync_func);
 }
 
 /**
